### PR TITLE
Feat: Adds tokens to `OutOfStock`

### PR DESCRIPTION
**Pull request by ArthurTriis1**
## What's the purpose of this pull request?
This PR applies the new [Theming structure](https://github.com/vtex-sites/base.store/pull/407) to `OutOfStck`: new scoped tokens linked to global ones.

Additionally, I adjusted the OutOfStock's style.

## How does it work?
This PR uses local variables to parameterize the `OutOfStock` properties, then connects these scoped tokens to the global ones.

Screen Shot 2022-06-06 at 14 34 16

### Title

<table><thead><tr><th>Local token</th><th>Default value/Global token linked</th></tr></thead><tbody><tr><td><code class="css-1npuvnl">--fs-out-of-stock-title-margin-bottom</code></td><td><code class="css-1npuvnl">var(--fs-spacing-0)</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-title-line-height</code></td><td><code class="css-1npuvnl">1.15</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-title-size</code></td><td><code class="css-1npuvnl">var(--fs-text-size-3)</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-title-weight</code></td><td><code class="css-1npuvnl">var(--fs-text-weight-bold)</code></td></tr></tbody></table>

### Message

<table><thead><tr><th>Local token</th><th>Default value/Global token linked</th></tr></thead><tbody><tr><td><code class="css-1npuvnl">--fs-out-of-stock-message-column-gap</code></td><td><code class="css-1npuvnl">var(--fs-spacing-0)</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-message-margin-bottom</code></td><td><code class="css-1npuvnl">var(--fs-spacing-3)</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-message-line-height</code></td><td><code class="css-1npuvnl">1.15</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-message-size</code></td><td><code class="css-1npuvnl">var(--fs-text-size-base)</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-message-weight</code></td><td><code class="css-1npuvnl">var(--fs-text-weight-regular)</code></td></tr></tbody></table>

### Button

<table><thead><tr><th>Local token</th><th>Default value/Global token linked</th></tr></thead><tbody><tr><td><code class="css-1npuvnl">--fs-out-of-stock-button-width</code></td><td><code class="css-1npuvnl">100%</code></td></tr><tr><td><code class="css-1npuvnl">--fs-out-of-stock-button-margin-top</code></td><td><code class="css-1npuvnl">var(--fs-spacing-3)</code></td></tr></tbody></table>

## Checklist
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 
- [x] Updated the Storybook - *if applicable*.
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.

### CODE REVIEW

File: src/components/product/OutOfStock/OutOfStock.tsx [hunk 1]
```diff
@@ -95,9 +96,13 @@ function OutOfStock(props: OutOfStockProps) {
   }
 
   return (
-    <UIOutOfStock onSubmit={handleSubmit}>
-      <UIOutOfStockTitle>{title}</UIOutOfStockTitle>
-      <UIOutOfStockMessage>
+    <UIOutOfStock
+      className={styles.fsOutOfStock}
+      onSubmit={handleSubmit}
+      data-fs-out-of-stock
```
Comment (hellofanny) [inline src/components/product/OutOfStock/OutOfStock.tsx:102]:
```suggestion
      data-fs-out-of-stock
      className={styles.fsOutOfStock}
      onSubmit={handleSubmit}
```

Comment (hellofanny) [reply src/components/product/OutOfStock/OutOfStock.tsx:102]: Maybe we can adopt this as a pattern, what do you think? Always place the `data-attr` first?

File: src/components/product/OutOfStock/out-of-stock.module.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
+@import "src/styles/scaffold";
+
+.fs-out-of-stock[data-fs-out-of-stock] {
```
Comment (hellofanny) [inline src/components/product/OutOfStock/out-of-stock.module.scss:3]:
```suggestion
.fs-out-of-stock {
```

We decided to use only the class here.

File: src/components/product/OutOfStock/out-of-stock.module.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  // --------------------------------------------------------
+  // Design Tokens for Out of Stock
+  // --------------------------------------------------------
+
+  // Title
+  --fs-out-of-stock-title-margin-bottom   : var(--fs-spacing-0);
+  --fs-out-of-stock-title-line-height     : 1.15;
+  --fs-out-of-stock-title-size            : var(--fs-text-size-3);
```
Comment (hellofanny) [inline src/components/product/OutOfStock/out-of-stock.module.scss:11]:
```suggestion
  --fs-out-of-stock-title-size            : var(--fs-text-size-lead);
```

File: src/components/product/OutOfStock/out-of-stock.module.scss [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  --fs-out-of-stock-title-weight          : var(--fs-text-weight-bold);
+
+  // Message
+  --fs-out-of-stock-message-column-gap    : var(--fs-spacing-0);
+  --fs-out-of-stock-message-margin-bottom : var(--fs-spacing-3);
+  --fs-out-of-stock-message-line-height   : 1.15;
+  --fs-out-of-stock-message-size          : var(--fs-text-size-base);
```
Comment (hellofanny) [inline src/components/product/OutOfStock/out-of-stock.module.scss:18]:
```suggestion
  --fs-out-of-stock-message-size          : var(--fs-text-size-body);
```

File: src/components/product/OutOfStock/OutOfStock.stories.mdx [hunk 5]
```diff
@@ -24,27 +44,88 @@ Out of Stock are used to announce to users that the product is out of stock. And
...
+    token="--fs-out-of-stock-title-margin-bottom"
+    value="var(--fs-spacing-0)"
+  />
+  <TokenRow
+    token="--fs-out-of-stock-title-line-height"
+    value="1.15"
+  />
+  <TokenRow
+    token="--fs-out-of-stock-title-size"
+    value="var(--fs-text-size-3)"
```
Comment (hellofanny) [inline src/components/product/OutOfStock/OutOfStock.stories.mdx:89]:
```suggestion
    value="var(--fs-text-size-lead)"
```

**Comment by hellofanny** [comment]
Niceeee! I left some suggestions 😃

File: src/components/product/OutOfStock/OutOfStock.tsx [hunk 6]
```diff
@@ -27,20 +23,25 @@ export interface OutOfStockProps {
   buttonText?: string
   /**
    * Icon displayed inside the button.
```
Comment (eduardoformiga) [inline src/components/product/OutOfStock/OutOfStock.tsx:25]: Looks like the comments are switched hahah.
```suggestion
   * Message describing when the user will be notified.
```

File: src/components/product/OutOfStock/OutOfStock.tsx [hunk 7]
```diff
@@ -27,20 +23,25 @@ export interface OutOfStockProps {
...
+   */
+  notificationMsg?: string
+  /**
+   * Icon displayed inside the message.
```
Comment (eduardoformiga) [inline src/components/product/OutOfStock/OutOfStock.tsx:29]:
```suggestion
   * Icon displayed inside the button.
```

File: src/components/product/OutOfStock/OutOfStock.tsx [hunk 8]
```diff
@@ -27,20 +23,25 @@ export interface OutOfStockProps {
...
    * @default <Icon name="BellRinging" />
    */
   buttonIcon?: ReactElement
   /**
    * Message describing when the user will be notified.
```
Comment (eduardoformiga) [inline src/components/product/OutOfStock/OutOfStock.tsx:34]:
```suggestion
   * Icon displayed inside the notification message.
```

**Review by eduardoformiga — COMMENTED** [review]
LGTM!  👏 

I noticed some description comments switched regarding the internal props for the component. Could you please take a look? 👀 

Screen Shot 2022-06-06 at 19 13 05

**Review by eduardoformiga — APPROVED** [review]

**Review by eduardoformiga — APPROVED** [review]

File: src/components/product/OutOfStock/out-of-stock.module.scss [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  // Title
+  --fs-out-of-stock-title-margin-bottom   : var(--fs-spacing-0);
+  --fs-out-of-stock-title-line-height     : 1.15;
+  --fs-out-of-stock-title-size            : var(--fs-text-size-lead);
+  --fs-out-of-stock-title-weight          : var(--fs-text-weight-bold);
+
+  // Message
+  --fs-out-of-stock-message-column-gap    : var(--fs-spacing-0);
+  --fs-out-of-stock-message-margin-bottom : var(--fs-spacing-3);
+  --fs-out-of-stock-message-line-height   : 1.15;
```
Comment (renatamottam) [inline src/components/product/OutOfStock/out-of-stock.module.scss:17]:
```suggestion
  --fs-out-of-stock-message-line-height   : var(--fs-out-of-stock-title-line-height);
```

File: src/components/product/OutOfStock/out-of-stock.module.scss [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  // Design Tokens for Out of Stock
+  // --------------------------------------------------------
+
+  // Title
+  --fs-out-of-stock-title-margin-bottom   : var(--fs-spacing-0);
+  --fs-out-of-stock-title-line-height     : 1.15;
+  --fs-out-of-stock-title-size            : var(--fs-text-size-lead);
+  --fs-out-of-stock-title-weight          : var(--fs-text-weight-bold);
+
+  // Message
```
Comment (renatamottam) [inline src/components/product/OutOfStock/out-of-stock.module.scss:14]: Could you add a color token for the message too? It should be using `--fs-color-success-text`

**Review by renatamottam — APPROVED** [review]
Great job! Left a couple suggestions :)

**Pull request merged**